source/shared/cpp/ObjectModel/ParseUtil.cpp

```diff
@@ -306,7 +306,7 @@ std::string ParseUtil::ToLowercase(std::string const &value)
 {
     std::string new_value;
     new_value.resize(value.size());
-    auto new_end = std::transform(value.begin(), value.end(), new_value.begin(), [](char c) { return std::tolower(c, std::locale()); });
```

Comment:
I actually added this one on purpose. This was part of PR #1579, and was introduced due to core guideline [ES.84](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es84-dont-try-to-declare-a-local-variable-with-no-name) - _"Don't (try to) declare a local variable with no name"_.

Per original PR:
> Basically, some functions return an object and some operations create temporary objects. These objects may have complex ctors or dtors which run in a way and at a time that's not obvious to the reader. 

> There are a few places where we were calling either `std::transform()` or an `insert()` function where it didn't make sense to change to a different form. For these, I assigned the return to a new local variable, which squelches the warning and also alerts the reader to the fact that there's an object being created (even if it's not going to be used).

Reply:
I see, and understand. Still curious how do you feel about it at a personal level? (As a programmer I’ve used unnamed objects and don’t agree that adding names makes anything other than clutter up.)
Again, agree to the value of following Core so will change back.

Reply:
I'm a little on the fence about it. In the specific instances you changed back, I agree that it introduces a bit of clutter. However, I think it's worth it to force future devs to recognize that there's a complex object being created. It also makes scoping clear (objects that construct and destruct in place are weird) :)